### ISSUE TITLE
Removing black bar when aspect=true and output=direct3d/opengl

### ISSUE THREAD

**Issue by Farooq87** [joncampbell123/dosbox-x#2374]
Please explain how to remove annoying black bar or make adjustment in source to fully stretch image to window using aspect=true. Its really convenient when this black bar is absent for people who prefer Dosbox-X in window mode. As for me I use auto transcriber which converts english speech to words from games in neighbor window. I must often pause dosbox to switch on transcriber window to pause it too, because it cant recognize prolonged speech at once, but 30 seconds max. At the same time I want to see toolbar fully.
Black bar in my case is just loss of a space atop.  
![bar](https://user-images.githubusercontent.com/80591773/111121243-4f625400-8564-11eb-91e2-42bd41d749bb.png)
If aspect = false then black bar disappears, but window automatically "stretches down" where I cant see it on my monitor
![bar2](https://user-images.githubusercontent.com/80591773/111121677-e0392f80-8564-11eb-8e90-ed9fcdca4668.png)

**Comment by joncampbell123** [comment]
You might be stretching the window down below the bottom of your taskbar. Does maximizing the window help?

When aspect correction is active, black bars on the top or sides is normal.

**Comment by Farooq87** [comment]
My dosbox window is on maximum. 
Well, when I used Daum 14-15 before switching on Dosbox-X there was possibility to scale on full size of monitor window without black bar with aspect=true.   
![bar3](https://user-images.githubusercontent.com/80591773/111124788-d285a900-8568-11eb-899c-77ff639847a4.png)

**Comment by Farooq87** [comment]
@JonCampbell123
My Host taskbar is attached to the side of the screen and hidden permanently using taskbar eliminator. Nice tool for folks who use to play in window mode

**Comment by Farooq87** [comment]
In fullscreen mode with aspect=true there are black bars on the side, but its ok. It makes rendering to proportion 4:3.  I just want no black bar in window mode if its possible please like in SVN Daum with aspect correction

**Comment by Farooq87** [comment]
@Joncampbell123 @Wengier
What exactly in source code is responsible for this feature?

**Comment by joncampbell123** [comment]
@Farooq87 There is common aspect correction code for the outputs in src/output/output_tools.h.

Just a thought: Are you using a fixed window size in dosbox.conf?

**Comment by Farooq87** [comment]
@joncampbell123
aspect = true +
![23522352](https://user-images.githubusercontent.com/80591773/111503129-63fa4400-873e-11eb-83a8-9ca4a017ce97.png)

**Comment by Wengier** [comment]
@Farooq87 Can you try a different setting for the ```windowresolution``` option, such as the default setting?

**Comment by Farooq87** [comment]
yep, default seems to decrease size of window to 3:4. In doing so I lose some space on the screen. Need somehow to stretch windown down to maximum and guest taskbar wouldnt disappear at the same time. Aspect = true gives such result in window, but black bar pops up on the top.
If only I could compare 2 source codes of Daum 15 and Dosbox-X  in src/output/output_tools.h. to see where is the difference.

**Comment by Farooq87** [comment]
@Wengier @joncampbell123
What exactly there need to be adjusted?
![aspect1](https://user-images.githubusercontent.com/80591773/111880746-1fbcad00-89a5-11eb-9310-52f2ffaa0d7b.png)
![aspect2](https://user-images.githubusercontent.com/80591773/111880750-24816100-89a5-11eb-9acf-9f5570d2c632.png)
